sdk/openai/openai/sources/customizations/api/operations.ts

```diff
@@ -198,28 +295,155 @@ function getChatCompletionsResult(body: Record<string, any>): Omit<ChatCompletio
 /** Convenience alias for BeginAzureBatchImageGenerationOptions */
 export type ImageGenerationOptions = BeginAzureBatchImageGenerationOptions;
 
+function _getChatCompletionsSendX(
+  context: Client,
+  messages: ChatMessage[],
+  deploymentName: string,
+  options: GetChatCompletionsOptions = { requestOptions: {} }
+): StreamableMethod<
+  | GetChatCompletionsWithAzureExtensions200Response
+  | GetChatCompletionsWithAzureExtensionsDefaultResponse
+> {
+  return options.azureExtensionOptions?.extensions
+    ? _getChatCompletionsWithAzureExtensionsSend(context, messages, deploymentName, {
+        ...options,
+        dataSources: options.azureExtensionOptions?.extensions,
+      })
+    : _getChatCompletionsSend(context, messages, deploymentName, options);
+}
+
 export function listChatCompletions(
   context: Client,
   messages: ChatMessage[],
   deploymentName: string,
   options: GetChatCompletionsOptions = { requestOptions: {} }
-): AsyncIterable<Omit<ChatCompletions, "usage">> {
-  const response = _getChatCompletionsSend(context, messages, deploymentName, {
+): AsyncIterable<ChatCompletions> {
+  const response = _getChatCompletionsSendX(context, messages, deploymentName, {
     ...options,
     stream: true,
   });
   return getOaiSSEs(response, getChatCompletionsResult);
 }
 
-export function listCompletions(
+export async function _getChatCompletionsDeserialize(
+  result: GetChatCompletions200Response | GetChatCompletionsDefaultResponse
+): Promise<ChatCompletions> {
+  if (isUnexpected(result)) {
+    throw result.body.error;
+  }
+
+  return {
```

Comment:
Since there are so many mappings here, could we move this into a separate deserializer file and have it unit testable?  Same goes for a lot of other mappings to ensure that we're capturing the right things

Reply:
Done in https://github.com/Azure/azure-sdk-for-js/pull/26907/commits/c27052ea439aa224df870352f33555d0d77a53d2.